### ISSUE TITLE
GHA Windows: `Parser error: while parsing a block mapping at line 1, column 1 did not find expected key at line 1, column 45`

### ISSUE THREAD

**Issue by bschilder** [neurogenomics/EpiCompare#146]
## 1. Bug description

Error occurs on GHA Windows:
https://github.com/neurogenomics/EpiCompare/actions/runs/4195716011/jobs/7275652659

```
Quitting from lines 193-210 (EpiCompare.Rmd) 
Error: Error: processing vignette 'EpiCompare.Rmd' failed with diagnostics:
Parser error: while parsing a block mapping at line 1, column 1 did not find expected key at line 1, column 45
```

I'm noticing "***" at line 193 (Which i think @serachoi1230 may have been using for Rmarkdown styling?). Perhaps Windows gets confused by these.

![Screenshot 2023-02-16 at 16 38 16](https://user-images.githubusercontent.com/34280215/219430036-0970a908-71de-47ca-bc79-1ef7ac249da4.png)

### Console output 

```
Loading required package: rcmdcheck
── R CMD build ─────────────────────────────────────────────────────────────────
* checking for file 'D:\a\EpiCompare\EpiCompare/DESCRIPTION' ... OK
* preparing 'EpiCompare':
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... ERROR
Error: --- re-building 'EpiCompare.Rmd' using rmarkdown
Quitting from lines 193-210 (EpiCompare.Rmd) 
Error: Error: processing vignette 'EpiCompare.Rmd' failed with diagnostics:
Parser error: while parsing a block mapping at line 1, column 1 did not find expected key at line 1, column 45
--- failed re-building 'EpiCompare.Rmd'
--- re-building 'docker.Rmd' using rmarkdown
--- finished re-building 'docker.Rmd'
--- re-building 'example_report.Rmd' using rmarkdown
--- finished re-building 'example_report.Rmd'
SUMMARY: processing the following file failed:
  'EpiCompare.Rmd'
Error: Error: Vignette re-building failed.
Execution halted
Error: Error in proc$get_built_file() : Build process failed
Calls: <Anonymous> ... build_package -> with_envvar -> force -> <Anonymous>
Execution halted
Error: Process completed with exit code 1.
Run actions/upload-artifact@v3
Warning: No files were found with the provided path: check. No artifacts will be uploaded.
```

### Expected behaviour   

Rmarkdown renders on all platforms.

## 2. Reproducible example   
 
https://github.com/neurogenomics/EpiCompare/actions/runs/4195716011/jobs/7275652659

**Comment by bschilder** [comment]
Just tried deleting the "***" and pushing

**Comment by bschilder** [comment]
> Just tried deleting the "***" and pushing

This didn't seem to do anything. In fact, the error is still indicating line 193. 

I noticed this also happens to be be just before the first header with `{.tabset}`. Possibly related?
https://github.com/neurogenomics/EpiCompare/actions/runs/4196426270/jobs/7277384263

**Comment by bschilder** [comment]
Possibly related errors regarding YAML header syntax:
https://github.com/rstudio/shiny/issues/1626

And/or Rmd file location:
https://github.com/rstudio/rmarkdown/issues/1005

**Comment by Tomrrr1** [comment]
I managed to narrow the error down to several `shQuote()` commands in `report_header.R`. `shQuote()` is used to quote strings in a way that is safe to pass to the command line. This function has a type argument which specifies the type of shell quoting. The default on Mac and Linux is "sh", whereas the Windows default is "cmd". Interestingly, `shQuote()` produces an error on Windows when using "cmd" despite this being the default. Even more interesting is that hard-coding the Mac and Linux default (type = "sh") fixes the Windows error! I have pushed this change to Bioconductor, and EpiCompare is now building on all systems. 

Summary: Problem with the `shQuote()` command in `report_header.R`. Manually specifying type = "sh" resolves the Windows bug. 

![Image](https://github.com/neurogenomics/EpiCompare/assets/76951517/ba78091c-4532-4b0d-8553-d0457dbb80e3)
